File: apps/scraper/examples/node-basic/scraper.js
/**
 * Example TimeScrape scraper — fetches events from a public API.
 */

import axios from "axios";
import { output } from "@timescrape/helper";

const response = await axios.get("https://date.nager.at/api/v3/PublicHolidays/2026/AT", { timeout: 30_000 });

for (const holiday of response.data) {
  output.writeRow({
    title: holiday.localName,
    date: holiday.date,
    location: "Austria",
    description: holiday.name ?? "",
    url: `https://date.nager.at/publicholiday/${holiday.date}/AT`,
  });
}

output.save();
console.log(`Scraped ${output.rowCount} events`);

File: apps/timescrape/examples/python-basic/scraper.py
"""Example TimeScrape scraper — fetches events from a public API."""

import requests
from timetiles.scraper import output

# Fetch sample data from a public API
response = requests.get("https://date.nager.at/api/v3/PublicHolidays/2026/DE", timeout=30)
response.raise_for_status()

holidays = response.json()

for holiday in holidays:
    output.write_row({
        "title": holiday["localName"],
        "date": holiday["date"],
        "location": "Germany",
        "description": holiday.get("name", ""),
        "url": f"https://date.nager.at/publicholiday/{holiday['date']}/DE",
    })

output.save()
print(f"Scraped {output.row_count} events")
